core/src/main/scala/org/apache/spark/deploy/worker/DriverWrapper.scala

```diff
@@ -66,4 +75,50 @@ object DriverWrapper {
         System.exit(-1)
     }
   }
+
+  // R or Python are not supported in cluster mode so just get jars and files for the driver
+  private def setupDependencies(loader: MutableURLClassLoader, userJar: String): Unit = {
+
+    var packagesExclusions = sys.props.get("spark.jars.excludes").orNull
+    var packages = sys.props.get("spark.jars.packages").orNull
+    var repositories = sys.props.get("spark.jars.repositories").orNull
+    val hadoopConf = new HadoopConfiguration()
+    val childClasspath = new ArrayBuffer[String]()
+    var jars = sys.props.get("spark.jars").orNull
```

Comment:
Ivy does the resolution and downloads them locally at the file system's node where the driver is launched on. So then you use the hadoop utilities to fetch them. That's the idea about jars. I was only focusing on packages... but yes need to write some more code to handle files and remote jars already passed though.